### PR TITLE
Waveforms: Disable textured waveforms when using OpenGL ES

### LINKED ISSUE #13380: Compile errors in `WaveformRendererTextured` when building for iOS (OpenGL ES 3.0)
### Bug Description

Parts of the new waveform renderer unfortunately no longer build for iOS (and possibly other platforms using OpenGL ES):

Screenshot 2024-06-16 at 16 45 51

We might have to investigate which parts to cond-compile out when using GL ES (e.g. testing for `QT_OPENGL_ES_2`).

### Version

https://github.com/fwcd/mixxx/commit/ea5e98073c15efb4c60f611fb67af457bb5be8ac (based on c368119262913411f94766a3e59735678bd72832)

### OS

macOS 14.5

### PR DESCRIPTION
**Pull request by fwcd**
Fixes #13380 (potential future refactoring moved to #13385)

The textured waveform renderer seems to use some OpenGL APIs that are unavailable in GL ES (e.g. on iOS), therefore this PR cond-compiles out `WaveformRendererTextured` when `QT_OPENGL_ES_2` is set.

### CODE REVIEW

**Comment by fwcd** [comment]
Perhaps it would make sense to disable the "High Details" checkbox when using GL ES? Given that it will have no effect

IMG_0904

Thoughts welcome.

**Comment by daschuer** [comment]
Disabling or even hiding the box is a good idea as a band aid. But I would like to keep the original bug open until we have figured out the root cause. I have edited the bug PR description accordingly.

**Comment by daschuer** [comment]
Please rebase to 2.5, since this is fixing 2.5-beta code.

**Comment by fwcd** [comment]
I'll rebase it once I'm home later.

> But I would like to keep the original bug open until we have figured out the root cause. 

The root cause is that the textured waveforms are implemented in terms of OpenGL APIs that are unavailable in OpenGL ES. So if we want to keep the original issue open, we might want to rephrase it to something along the lines of "Reimplement textured waveforms in OpenGL ES subset" or similar, since this specific issue is fixed.

**Comment by fwcd** [comment]
I don't think it makes sense to target 2.5 here currently, since it is based on the refactorings introduced in #13220 (`addWaveformSignalRenderer` etc.) that are only in `main`. It might be possible to backport, but in doing so we'll get merge conflicts and I'd base iOS support on `main` anyway (given that it's highly experimental).

**Review by acolombier — APPROVED** [review]
LGTM! Thanks for the fix

**Review by daschuer — APPROVED** [review]
LGTM, thank you.
